File: 5-longest-palindromic-substring/longest-palindromic-substring.cpp
class Solution {
public:
    string longestPalindrome(string s) {

        int resLen=0;
        int resInd=0;

    for(int i=0;i<s.size();i++)
    {
        //for the odd len string you can use the logic of increasing from the middle.

        int start=i;
        int end=i;

        //odd length string


        while(start>=0 && end<s.size() && s[start]==s[end])
        {
            if(resLen<end-start+1)
            {
                resLen=end-start+1;
                resInd=start;
            }

            start--;
            end++;


        }
//even length string
        start=i;
        end=i+1;

        while(start>=0 && end<s.size() && s[start]==s[end])
        {
            if(resLen<end-start+1)
            {
                resLen=end-start+1;
                resInd=start;
            }

            start--;
            end++;


        }


    }

    return s.substr(resInd, resLen);
        
    }
};

File: 5-longest-palindromic-substring/longest-palindromic-substring.py
class Solution:
    def longestPalindrome(self, s: str) -> str:

        maxL=0
        res=[-1,-1]
        
        for i in range(len(s)):

            l=i
            r=i


            while l>=0 and r<len(s) and (s[l]==s[r]):

                if r-l+1>maxL:
                    res=[l,r]
                    maxL=max(maxL,r-l+1)


                l-=1
                r+=1

            l=i
            r=i+1

            while l>=0 and r<len(s) and (s[l]==s[r]):
                if r-l+1>maxL:
                    res=[l,r]
                    maxL=max(maxL,r-l+1)
                l-=1
                r+=1
        l,r=res
        return s[l:r+1]

        